schema/m3-requirements.schema.json

```diff
@@ -640,6 +704,30 @@
             "type": "integer",
             "title": "Item Not Collected",
             "description": "Fulfilled by an item at a given node not having been collected yet."
+          },
+          "gainFlashSuit": {
+            "$id": "#/definitions/logicalRequirement/items/properties/gainFlashSuit",
+            "type": "object",
+            "title": "Gain Flash Suit",
+            "description": "Always fulfilled, with an effect of Samus gaining a flash suit.",
+            "additionalProperties": false,
+            "properties": {}
+          },
+          "useFlashSuit": {
+            "$id": "#/definitions/logicalRequirement/items/properties/useFlashSuit",
+            "type": "object",
+            "title": "Use Flash Suit",
+            "description": "Fulfilled if Samus has a flash suit, with an effect of losing the flash suit.",
+            "additionalProperties": false,
+            "properties": {}
+          },
+          "noFlashSuit": {
+            "$id": "#/definitions/logicalRequirement/items/properties/noFlashSuit",
+            "type": "object",
+            "title": "No Flash Suit",
+            "description": "Fulfilled if Samus does not have a flash suit.",
```

Comment:
does it assume Samus can always get rid of a flash suit for free? (assuming its probably fine)

Reply:
The way it's worded here wouldn't necessarily assume that, but I think we will assume that in the randomizer implementation, as it should make things more efficient while being logically equivalent as far as I can tell. It's not that you can necessarily always get rid of a flash suit for free; it's that you could have chosen to have not gotten it in the first place. This should be logically sound as long as we don't add strats that unavoidably produce a flash suit as a side effect of something else (and without a free way to get rid of it); I'm not aware of any strats like that, but if it comes up then we can think about it some more. 

When we get to blue suits, I think superjumps are an example where you can get a blue suit as a side effect of something else (whether you wanted it or not); but a blue suit is trivial to get rid of by just pressing dash so I think it should also be fine in that case.